x/mongo/driver/ocsp/ocsp.go

```diff
@@ -91,6 +104,43 @@ func Verify(ctx context.Context, connState tls.ConnectionState) error {
 	return nil
 }
 
+// getParsedResponse attempts to parse a response from the stapled OCSP data or by contacting OCSP responders if no
+// staple is present.
+func getParsedResponse(ctx context.Context, cfg config, connState tls.ConnectionState) (*Response, error) {
+	cachedResponse := cfg.cache.Get(cfg.ocspRequest)
+	stapledResponse, err := parseStaple(cfg, connState.OCSPResponse)
+	if err != nil {
+		return nil, err
+	}
+
+	if stapledResponse != nil {
+		// If there is a staple, attempt to cache it. The cache.Put call will resolve conflicts with an existing cache
+		// enry if necessary.
+		return cfg.cache.Put(cfg.ocspRequest, newResponse(stapledResponse)), nil
+	}
+	if cachedResponse != nil {
+		return cachedResponse, nil
+	}
+
+	// If there is no stapled or cached response, fall back to querying the responders if that functionality has not
+	// been disabled.
+	if cfg.disableEndpointChecking {
+		return nil, nil
+	}
+	externalResponse, err := contactResponders(ctx, cfg)
+	if err != nil {
+		return nil, err
+	}
+	if externalResponse == nil {
+		// None of the responders were available.
+		return nil, nil
+	}
+
+	// Similar to the stapled response case above, unconditionally call Put and it will resolve conflicts with existing
+	// cache entries.
```

Comment:
This comment is confusing as there should be no cached response, as it would have been returned at line 122.  Is it because of a race with something else updating the cache?

Reply:
Yes, it is because there could have been a race with something else updating the cache in the meantime, but I can see how this is confusing. Updated the comment.